src/Kiota.Builder/Configuration/GenerationConfiguration.cs

```diff
@@ -197,6 +198,7 @@ public ApiDependency ToApiDependency(string configurationHash, Dictionary<string
         {
             dependency.Extensions.Add(KiotaHashManifestExtensionKey, JsonValue.Create(configurationHash));// only include non empty value.
         }
+        dependency.Extensions.Add(KiotaVersionManifestExtensionKey, Kiota.Generated.KiotaVersion.Current());
```

Comment:
don't we need to parse it as well when running updates so it's used for the comparison?

Reply:
You mean in the extension? That's the next phase I'm working on. Still trying to figure out how to accommodate the multiple version locations.